### DISCUSSION TITLE
"ambiguous column name" during join

### DISCUSSION THREAD

**Discussion by myslak71** [sqlalchemy/sqlalchemy#6324] [Usage Questions]
Hey there!

I have a following code:
```
class Parent(db.Model):
    __tablename__ = "parents"

    id = db.Column(db.String(255), primary_key=True)
    name = db.Column(db.String(100), nullable=False, server_default="")
    children = db.relationship("Child", back_populates="parent")


class Child(db.Model):
    __tablename__ = "children"

    id = db.Column(db.String(255), primary_key=True)
    name = db.Column(db.String(100), nullable=False, server_default="")
    parent_id = db.Column(db.Integer, db.ForeignKey("parents.id"))
    parent = db.relationship("Parent", back_populates="children")

query = (
    Parent.query.options(joinedload(Parent.children))
    .order_by(asc(text("name")))
)
print(query.all())
```

and it produces an error:
```
sqlite3.OperationalError: ambiguous column name: name
```

I know it happens because both tables have a `name` column and this can be fixed by adding replacing with `order_by(asc(text("parents.name")))` or `order_by(asc(Parent.name))`.
But it is doable by labeling/aliasing in the query? I don't want to change this part `asc(text("name"))`

**Comment by zzzeek** [comment]
hey there -

Yes, SQLAlchemy has a special feature in ORDER BY / GROUP BY where you can pass string names of columns that will be matched up to that name in the SELECT:

```
query = (
    s.query(Parent).options(joinedload(Parent.children))
    .order_by(asc("name"))
)

```

there is a definite issue in the documentation that this is a detail that is buried in the new tutorial, but all the other order_by docs don't link to it at all, but here's the doc: https://docs.sqlalchemy.org/en/14/tutorial/data.html#ordering-or-grouping-by-a-label   this also works in 1.3 where at the moment I am not seeing it docuemtned anywhere at all, that's a problem.  it's likely in one of the migration guides but that's not good enough i will add some linkage now.

**Reply by zzzeek** [reply]
oh here it is in 1.3...more links needed

https://docs.sqlalchemy.org/en/14/core/tutorial.html#ordering-or-grouping-by-a-label
